app/main.py

```diff
@@ -2,24 +2,68 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: type, owner: type) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: type, value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
+        else:
+            raise ValueError(f"{self.protected_name}"
+                             f" must be lesser than {self.min_amount}"
+                             f" and more than {self.max_amount}")
```

Comment:
The error message here can be misleading. It says the value must be lesser than the minimum and more than the maximum, which is not possible. It should be 'value must be greater than or equal to the minimum and lesser than or equal to the maximum'.